### PR TITLE
rom import by system

### PR DESCRIPTION
**Pull request by blathers16**
This PR implements the ability to import only ROMs for one system at a time.  There are probably some issues with it, but let me know what needs to be fixed and I will get right on it.  In particular, I wasn't really sure what I was doing with the dialog script.  I don't know if you expect tests for this, if so, please let me know what you expect them to test in this case.

### CODE REVIEW

**Comment by blathers16** [comment]
Hi, I noticed that the menu has been reworked in testing, what do I need to do to update this to match the new menu setting?  Do I just need to split the entries into separate files?  Are any other changes needed?

**Comment by sairuk** [comment]
i'll take a look but yes as you have noted this pull request doesn't implement the new menu system changes and will need to be reworked to fit.

Each menu has its own file now then is a step towards modular menus later on

File: dialog/romimportsystem.sh [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  do
+    source $_CONFIG
+    
+    case ${CHOICE} in
+      01)
+        EXIT_OK
+      ;;
+      02)
+        CLEAR
+        /opt/retronas/scripts/romimport.sh -t ${SYSTEM_ARR[2]}
```
Comment (sairuk) [inline dialog/romimportsystem.sh:46]: This should be able to be moved to a function to reduce duplication throughout the file and call it with the menu entry param

File: dialog/romimportsystem.sh [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+rn_import_system() {
+
+  local MENU_NAME=romimportsystem
+  READ_MENU_JSON "${MENU_NAME}"
+  READ_MENU_TDESC "${MENU_NAME}"
+  DLG_MENUJ "${MENU_TNAME}" 10 "${MENU_BLURB}"
+  
+
+    while read -r line; do
+        lineArr=($line)
```
Comment (sairuk) [inline dialog/romimportsystem.sh:21]: we try to use uppercase variable names

File: dialog/romimportsystem.sh [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        STR=${lineArr[0]}
+        SUB="id"
+        if [[ "$STR" == *"$SUB"* ]]; then
+            SYSTEMS+=${lineArr[1]}' '
+        fi
+    done < <(/opt/retronas/scripts/romimport.sh -l)
+
+    i=2
```
Comment (sairuk) [inline dialog/romimportsystem.sh:29]: same comment on uppercase var names

File: config/retronas.json [hunk 4]
```diff
@@ -1552,9 +1552,9 @@
                     "index":"04",
                     "title": "ROM import",
                     "description": "Import roms via Smokemonster SMDBs",
-                    "id":"romimport",
+                    "id":"romimport-menu",
                     "prompt": "Open",
-                    "type": "dialog_yn",
```
Comment (sairuk) [inline config/retronas.json:1557]: we'll need to work on this menu, its pretty small on some term sizes so its not exactly apparent whats going on

**Comment by sairuk** [comment]
So I think updating your branch with the latest testing changes and then refitting your menu updates and script changes mindful of my comments then redo a new pull request will be best

i do have concerns on the additional overhead in the menu json for the system names, we can revisit this however at a later date

**Pull request closed without merging**